Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -151,12 +151,39 @@ class UpstreamIntegrationTests: XCTestCase {
         wait(for: [edgeRequestContentExpectation, networkResponseExpectation], timeout: asyncTimeout)
     }
     
-    func test_testFailureExample() {
-        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"],
-                                              data: ["data": ["test": "data"]])
+    func testJSONComparisonSystem() {
+        let multilineValidation = #"""
+          {
+            "integerCompare": 456,
+            "decimalCompare": 123.123,
+            "stringCompare": "abc",
+            "boolCompare": true,
+            "nullCompare": null,
+            "arrayCompare": [1,2,3],
+            "dictionaryCompare": {
+              "nested1": "value1"
+            }
+          }
+        """#
+        
+        let multilineInput = #"""
+          {
+            "integerCompare": 0,
+            "decimalCompare": 4.123,
+            "stringCompare": "def",
+            "boolCompare": false,
+            "nullCompare": "not null",
+            "arrayCompare": [1,2],
+            "dictionaryCompare": {
+              "nested1different": "value1"
+            }
+          }
+        """#
+
+        let jsonValidation = try? JSONDecoder().decode(JSON.self, from: multilineValidation.data(using: .utf8)!)
+        let jsonInput = try? JSONDecoder().decode(JSON.self, from: multilineInput.data(using: .utf8)!)
         
-        XCTAssertEqual("success", "not success")
-        XCTFail("This is an example failure message: \(experienceEvent)")
+        XCTAssertEqual(jsonValidation, jsonInput)
```

Comment:
I like the simplicity of this line, however currently the debugging experience is not great as the actual errors are not displayed on this line, but inside JSONUtils. 
Can we propagate the file and line where the error occurred back to the caller function so they are displayed in context?
See [XCTAssert(_:_:file:line:)](https://developer.apple.com/documentation/xctest/1500669-xctassert) and some examples in the [FunctionalTestBase](https://github.com/adobe/aepsdk-edge-ios/blob/main/Tests/FunctionalTests/util/FunctionalTestBase.swift#L260)

Reply:
+1 for moving them in a separate file as you will use those in multiple test files